armi/interfaces.py

```diff
@@ -77,6 +81,141 @@ class STACK_ORDER:  # pylint: disable=invalid-name, too-few-public-methods
+        """
+        Stores the previous iteration value of the given parameter.
+
+        Parameters
+        ----------
+        val : _SUPPORTED_TYPES
+            the value to store. Is commonly equal to interface.getTightCouplingValue()
+
+        Raises
+        ------
+        TypeError
+            Checks the type of the val against ``_SUPPORTED_TYPES`` before storing.
+            If invalid, a TypeError is raised.
+        """
+        if type(val) not in self._SUPPORTED_TYPES:
+            raise TypeError(
+                f"{val} supplied has type {type(val)} which is not supported in {self}. "
+                f"Supported types: {self._SUPPORTED_TYPES}"
+            )
+        self._previousIterationValue = val
+
+    def isConverged(self, val: _SUPPORTED_TYPES):
+        """
+        Return True if the convergence criteria between the current and previous iteration values are met.
+
+        Parameters
+        ----------
+        val : _SUPPORTED_TYPES
+            the value to store. Is commonly equal to interface.getTightCouplingValue()
+
+        Notes
+        -----
+        - On convergence, this class is automatically reset to its initial condition to avoid retaining
+          or holding a stale state. Calling this method will increment a counter that when exceeded will
+          clear the state. A warning will be reported if the state is cleared prior to the convergence
+          criteria being met.
+        - For computing convergence of arrays, only up to 2D is allowed. 3D arrays would arise from considering
+          component level parameters. However, converging on component level parameters is not supported at this time.
+
+        Raises
+        ------
+        ValueError
+            If the previous iteration value has not been assigned. The ``storePreviousIterationValue`` method
+            must be called first.
+        TypeError
+            If the type of the current value provided is not the same as the previous iteration value.
+        RuntimeError
+            Only support calculating norms for up to 2D arrays.
+        """
+        if self._previousIterationValue is None:
+            raise ValueError(
+                f"Cannot check convergence of {self} with no previous iteration value set. "
+                f"Set using `storePreviousIterationValue` first."
+            )
+
+        if not isinstance(val, type(self._previousIterationValue)):
+            raise TypeError(
+                f"The current value {val} is not the same type as the previous "
+                f"iteration value of {self._previousIterationValue}"
+            )
+
+        previous = self._previousIterationValue
+
+        # calculate convergence of val and previous
+        if isinstance(val, (int, float)):
+            self.eps = abs(val - previous)
+        else:
+            if len(numpy.array(val).shape) == 1:
+                self.eps = norm(numpy.subtract(val, previous), ord=2)
```

Comment:
Yeah, this will definitely be included in the docs. I just want to get general buy in on the PR as a whole before I start cranking on the docs. 

I'd also like to make the choice of norm exposed to the user. But that will be a future PR.

Reply:
I made an issue to track that desire: https://github.com/terrapower/armi/issues/1089